Sources/NIOCore/Interfaces.swift

```diff
@@ -87,6 +87,63 @@ public final class NIONetworkInterface {
     /// The index of the interface, as provided by `if_nametoindex`.
     public let interfaceIndex: Int
 
+#if os(Windows)
+    internal init?(_ pAdapter: UnsafeMutablePointer<IP_ADAPTER_ADDRESSES>,
+                   _ pAddress: UnsafeMutablePointer<IP_ADAPTER_UNICAST_ADDRESS>) {
+        self.name = String(decodingCString: pAdapter.pointee.FriendlyName,
+                           as: UTF16.self)
+        guard let address = pAddress.pointee.Address.lpSockaddr.convert() else {
+            return nil
+        }
+        self.address = address
+
+        // TODO: convert the prefix length to the mask itself
+        let v4mask: (UINT8) -> SocketAddress? = { _ in
+            var buffer: [CChar] =
+                Array<CChar>(repeating: 0, count: Int(INET_ADDRSTRLEN))
+            var mask: sockaddr_in = sockaddr_in()
+            mask.sin_family = ADDRESS_FAMILY(AF_INET)
+            _ = buffer.withUnsafeMutableBufferPointer {
+                try! NIOBSDSocket.inet_ntop(addressFamily: .inet,
+                                            addressBytes: &mask,
+                                            addressDescription: $0.baseAddress!,
+                                            addressDescriptionLength: socklen_t(INET_ADDRSTRLEN))
+            }
+            return SocketAddress(mask)
+        }
```

Comment:
Ah yeah, I remember this from last time: these functions don't work as implemented. I'll merge this regardless, and follow-up with correct implementations.

Reply:
Thanks!